[Lateral Movement Detection] Update package to add RDP based lateral movement detection

sodhikirti07 opened the pull request:
<!-- Type of change
Please label this PR with one of the following labels, depending on the scope of your change:
- Bug
- Enhancement
- Breaking change
- Deprecation
-->

## What does this PR do?
- Adds anomaly detection jobs and security rules to detect malicious RDP activity in the environment.
- Adds a folder that installs pivot transform in the user environment. The transform is created following the new guidelines for `8.8` ([link](https://docs.google.com/document/d/1dpqFxdTrX9ytOvMKHSZzXYUecxqqYu7Sb9_xLGKHajM/edit?usp=sharing))
- Update dashboard to include all the anomaly detection jobs. 
- Update README to include additional instructions, AD jobs and security rules.
- Updates changelog.yml and manifest.yml

<!-- Mandatory
Explain here the changes you made on the PR.
-->

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

<!-- Recommended
Add a checklist of things that are required to be reviewed in order to have the PR approved
-->
- [ ]

## How to test this PR locally

Stand the `elastic-package` stack using `8.8.0-SNAPSHOT` using the below command. This is for testing the transform with the latest guidelines and features. Note that previous versions of elastic with `elastic-package` will throw installation error for the transform. For more context, read the section for `_meta.run_as_kibana_system` [here](https://docs.google.com/document/d/1dpqFxdTrX9ytOvMKHSZzXYUecxqqYu7Sb9_xLGKHajM/edit#heading=h.dvxhz08g6b37) 

                         `elastic-package stack up -d -v --version 8.8.0-SNAPSHOT` 

<!-- Recommended
Explain here how this PR will be tested by the reviewer: commands, dependencies, steps, etc.
-->

## Related issues

<!-- Recommended
Link related issues below. Insert the issue link or reference after the word "Closes" if merging this should automatically close it.

- Closes #123
- Relates #123
- Requires #123
- Supersedes #123
-->
-

## Screenshots

<!-- Optional
Add here screenshots presenting:
- Kibana UI forms presenting configuration options exposed by the integration
- dashboards with collected metrics or logs
-->
Screenshots are in the comment section below.

sodhikirti07 commented:
**Screenshots**

- Landing page

![image](https://github.com/elastic/integrations/assets/109447885/1b2abcef-58e0-410b-b955-8805e32dee60)

- Transform installed and running

![image](https://github.com/elastic/integrations/assets/109447885/83ebf2f1-e0a2-40f8-ba68-dc3c8b85b5ca)

![image](https://github.com/elastic/integrations/assets/109447885/eea6f5b1-f4fe-4a55-85c6-f09993bebf3c)

- ML jobs started and retrieving data 

![image](https://github.com/elastic/integrations/assets/109447885/b36bd4ec-4c99-47e7-99ef-dd5d0a11971d)

![image](https://github.com/elastic/integrations/assets/109447885/e45ccf70-0808-43b0-96ba-394b4dcd990d)

![image](https://github.com/elastic/integrations/assets/109447885/0ccfd8fd-3aea-45ab-978f-c49f34b21d24)

- Security rules enabled 
  
![image](https://github.com/elastic/integrations/assets/109447885/2d14e507-8dc7-451f-a401-7592f549a227)

- Updated dashboard

![image](https://github.com/elastic/integrations/assets/109447885/3882050a-6c31-4cc7-91aa-0c9cf36b7346)

elasticmachine commented:
## :broken_heart: Build Failed


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-6406/detail/PR-6406/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-6406/detail/PR-6406/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-6406/detail/PR-6406/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-6406/detail/PR-6406/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://integrations_6406.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/fleet-ci/transactions/view?rangeFrom=2023-05-31T15:01:19.742Z&rangeTo=2023-05-31T15:21:19.742Z&transactionName=BUILD+Ingest-manager%2Fintegrations%2FPR-%7Bnumber%7D&transactionType=job&latencyAggregationType=avg&traceId=1236b956c89ee7dcb9d5ddb8f2b4d119&transactionId=487f437a19f18eb5)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2023-05-31T15:11:19.742+0000


* Duration: 14 min 13 sec





</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY -->














### Steps errors [![2](https://img.shields.io/badge/2%20-red)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-6406/detail/PR-6406/1//pipeline)
  <details><summary>Expand to view the steps failures</summary>
  <p>


  
  
  
  
##### `Test integration: lmd`
<ul>
<li>Took 0 min 2 sec . View more details <a href="https://fleet-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Ingest-manager/pipelines/integrations/pipelines/PR-6406/runs/1/steps/1149/log/?start=0">here</a></li>
<li>Description: <code>                                eval "$(../../build/elastic-package stack shellinit)"                                ../../build/elastic-package test -v --report-format xUnit --report-output file --test-coverage                                </code></l1>
</ul>
  
  
  
  
##### `Google Storage Download`
<ul>
<li>Took 0 min 0 sec . View more details <a href="https://fleet-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Ingest-manager/pipelines/integrations/pipelines/PR-6406/runs/1/steps/1238/log/?start=0">here</a></li>

</ul>
  
  </p>
  </details>



## :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

To re-run your PR in the CI, just comment with:



- `/test` : Re-trigger the build.




</p>
</details>

<!--COMMENT_GENERATED_WITH_ID_comment.id-->

qn895 commented on packages/lmd/docs/README.md:
```diff
@@ -9,6 +9,13 @@ To download the assets, click **Settings** > **Install Lateral Movement Detectio
 
 ### Add preconfigured anomaly detection jobs
 
+The anomaly detection jobs under this package rely on two indices. One has file transfer events (`logs-*`) and another index (`ml-rdp-lmd-1.0.2`) that collects RDP session information from a transform. Before enabling the jobs, create a custom index with both index patterns.
```
Should this be `create a data view with both index patterns`

sodhikirti07 replied:
Yes, thanks for catching this!

qn895 commented on packages/lmd/docs/README.md:
```diff
@@ -9,6 +9,13 @@ To download the assets, click **Settings** > **Install Lateral Movement Detectio
+1. Go to **Stack Management > Kibana > Data Views** and click **Create data view**.
+2. Enter the name of your respective index patterns in the **Index pattern** box, i.e., `logs-*, ml-rdp-lmd-1.0.2`, and copy the same in the **Name** field.
+3. Select `@timestamp` under the **Timestamp** field and click on **Save data view to Kibana**.
+4. Use the new custom index (`logs-*, ml-rdp-lmd-1.0.2`) to create anomaly detection jobs for this package.
```
Similarly, `use the new custom data view`

qn895 commented on packages/lmd/manifest.yml:
```diff
@@ -1,16 +1,16 @@
   - advanced_analytics_ueba
 conditions:
-  kibana.version: "^8.5.0"
+  kibana.version: "^8.7.0"
```
Looks like currently, the transform will fail @ 8.7 due to permission issue? If so, might be good to upgrade `kibana.version` to ^8.8.0.

sodhikirti07 replied:
Thank you!

ajosh0504 commented on packages/lmd/docs/README.md:
```diff
@@ -9,6 +9,13 @@ To download the assets, click **Settings** > **Install Lateral Movement Detectio
+4. Use the new data view (`logs-*, ml-rdp-lmd-1.0.2`) to create anomaly detection jobs for this package.
+
+
 In **Machine Learning > Anomaly Detection**, when you create a job, you should see an option to `Use preconfigured jobs` with a card for Lateral Movement Detection. When you select the card, you will see a pre-configured anomaly detection job that you can enable depending on what makes the most sense for your environment.
```
```suggestion
In **Machine Learning > Anomaly Detection**, when you create a job, you should see an option to `Use preconfigured jobs` with a card for Lateral Movement Detection. When you select the card, you will see pre-configured anomaly detection jobs that you can enable depending on what makes the most sense for your environment.
```

ajosh0504 commented on packages/lmd/elasticsearch/transform/pivot_transform/transform.yml:
```diff
@@ -0,0 +1,88 @@
+        - term:
+            process.Ext.session_info.logon_type:
+              value: RemoteInteractive
+  runtime_mappings:
```
What do you need this runtime mapping for? You don't seem to be using it anywhere in the aggregations, and you're filtering on the existence of the `host.ip` field anyway

ajosh0504 commented on packages/lmd/kibana/ml_module/lmd-ml.json:
```diff
@@ -169,6 +152,245 @@
+                    "analysis_config": {
+                        "bucket_span": "3h",
+                        "detectors": [
+                              {
```
What's the implication of having these different detector configurations in one job vs separate jobs for each of these detectors? If you haven't already, would be good to get confirmation from the Platform team on this.

sodhikirti07 replied:
Could you elaborate on the implications in term of what? Performance or something else?

ajosh0504 replied:
Performance, yes, but in the way anomalies at detected. Would this multi-detector configuration in a single job vs separate jobs for each detector produce the same anomalies? If not, what's the difference?

sodhikirti07 replied:
They do generate same anomalies when combined or used separately (I tested using both configs). The way these detectors work is like an `OR` statement. But I might check if there are other performance constraints like model size or memory, etc.

ajosh0504 commented on packages/lmd/kibana/ml_module/lmd-ml.json:
```diff
@@ -2,37 +2,20 @@
-                  "event.action": "creation"
-                }
+                        }},
+              {"exists": {"field": "session.start_time"}},
```
Can there be `session.start_time` without `session.end_time` and vice versa? If not, then you can drop one of them. If you strictly need to check for the presence of both, you will need to create a must clause encompassing both those checks inside this should clause.

sodhikirti07 replied:
Good catch! I need both of them, will add `must` for both of these.

ajosh0504 commented on packages/lmd/elasticsearch/transform/pivot_transform/transform.yml:
```diff
@@ -0,0 +1,88 @@
+            source: "if (doc['host.ip'].size()!=0){emit(doc['host.ip'][1]);} else{emit('None');}"
+dest:
+  index: "ml-rdp-lmd-1.0.2"
+description: This transform runs hourly and collects windows RDP session information for Lateral Movement Detection package.
```
How are you defining a session here?

sodhikirti07 replied:
So, a session includes all the activity a user/host does from remote signing in to signing out of the remote machine. Is this the definition you're looking for?

ajosh0504 commented on packages/lmd/kibana/ml_module/lmd-ml.json:
```diff
@@ -169,6 +152,245 @@
+                    ],
+                    "description": "A machine learning job to detect abnormally long RDP session duration.",
+                    "analysis_config": {
+                        "bucket_span": "3h",
```
How did you decide upon 3 hours? If the transform is only running every hour, 3 hours would only give you three values to calculate mean and variance for. 6h might be better here?

sodhikirti07 replied:
You are correct, I might need to adjust either the frequency of transform or the bucket_span.

ajosh0504 commented on packages/lmd/kibana/ml_module/lmd-ml.json:
```diff
@@ -169,6 +152,245 @@
+                    ],
+                    "description": "A machine learning job to detect unusually high number of processes started in a single RDP session.",
+                    "analysis_config": {
+                        "bucket_span": "3h",
```
Curious why you went with a bucket span of 3h here? I understand a longer bucket span for mean and variance calculations.

ajosh0504 commented on packages/lmd/kibana/ml_module/lmd-ml.json:
```diff
@@ -169,6 +152,245 @@
+                                "detector_index": 0
+                              },
+                              {
+                                "detector_description": "high_sum(number_processes_per_session) partitionfield=\"destination.ip\"",
```
What's the thought process behind looking for anomalous number of processes on source as well as destination IP?

ajosh0504 commented on packages/lmd/kibana/ml_module/lmd-ml.json:
```diff
@@ -169,6 +152,245 @@
+                    ],
+                    "description": "A machine learning job to detect an RDP session started at an usual time or weekday.",
+                    "analysis_config": {
+                        "bucket_span": "3h",
```
Curious about the bucket span here as well.

ajosh0504 commented on packages/lmd/kibana/ml_module/lmd-ml.json:
```diff
@@ -169,6 +152,245 @@
+                          {
+                            "detector_description": "time_of_day partitionfield=\"source.ip\"",
+                            "function": "time_of_day",
+                            "partition_field_name": "source.ip",
```
Wouldn't you want to detect unusual times for a particular user rather than source IP, since the entities we're mainly trying to model are users and hosts?

ajosh0504 commented on packages/lmd/kibana/ml_module/lmd-ml.json:
```diff
@@ -169,6 +152,245 @@
+                                "detector_description": "high_mean(session.duration) partitionfield=\"source.ip\"",
+                                "function": "high_mean",
+                                "field_name": "session.duration",
+                                "partition_field_name": "source.ip",
```
Wouldn't you want to detect these anomalies by users instead, since the entities we're trying to model are mainly users and hosts?

ajosh0504 commented on packages/lmd/kibana/ml_module/lmd-ml.json:
```diff
@@ -169,6 +152,245 @@
+                            "detector_index": 0
+                          }
+                            ],
+                        "influencers": [
```
Might want to include destination IP as an influencer as well.

ajosh0504 commented on packages/lmd/kibana/ml_module/lmd-ml.json:
```diff
@@ -169,6 +152,245 @@
+                                "detector_index": 0
+                              }
+                            ],
+                        "influencers": [
```
Might want to include source IP as an influencer as well.

ajosh0504 commented on packages/lmd/kibana/ml_module/lmd-ml.json:
```diff
@@ -169,6 +152,245 @@
+                        "bucket_span": "3h",
+                        "detectors": [
+                            {
+                              "detector_description": "high_mean(total_length_process_args) partitionfield=\"source.ip\"",
```
Similar q as above re: including detectors for both source and destination IP.

ajosh0504 commented on packages/lmd/kibana/security_rule/462dfff5-6d66-4489-9d1d-f3b352b83665.json:
```diff
@@ -0,0 +1,26 @@
+            "Elastic"
+        ],
+        "description": "An anomaly detection job has detected a high count of destination IPs establishing an RDP connection with a single source IP.\n",
+        "from": "now-210m",
```
How did you decide on this time range?

sodhikirti07 replied:
I kept it in coherence with the model `buclet_span`. Sometimes, the detection rules do not alert when their lookback time is < `bucket_span` of the model.

peteharverson replied:
The discussion on this [issue](https://github.com/elastic/kibana/issues/158152#issuecomment-1576468478) might be relevant here, for alerting rules we generally use a lookback interval of 2 x `bucket_span` plus the `query_delay` (where the `query_delay` is roughly 2s if not supplied in the datafeed config).

sodhikirti07 replied:
Done! Changed the lookback time.

ajosh0504 commented on packages/lmd/kibana/security_rule/4d9880ab-9843-474a-9c5a-13fc057a334b.json:
```diff
@@ -0,0 +1,26 @@
+        ],
+        "description": "Identifies potential malicious activity by spotting multiple hosts in an RDP session.\n",
+        "language": "kuery",
+        "from": "now-10m",
```
How did you decide on this time range?

ajosh0504 commented on packages/lmd/kibana/security_rule/4d9880ab-9843-474a-9c5a-13fc057a334b.json:
```diff
@@ -0,0 +1,26 @@
+        "license": "Elastic License",
+        "name": "Multiple hosts identified in an RDP session",
+        "query": "process.Ext.authentication_id :* and process.Ext.session_info.client_address:* and host.ip:* and process.Ext.session_info.logon_type:\"RemoteInteractive\"  and event.category :\"process\"",
+        "threshold": {"field":["process.Ext.authentication_id"],"value":2,"cardinality":[{"field":"host.name","value":2}]},
```
Will this trigger if 2 or more hosts are identified or strictly 2?

sodhikirti07 replied:
It will trigger if two or more hosts are detected in a single session

ajosh0504 commented on packages/lmd/kibana/security_rule/832da588-53eb-459b-8e1f-45d2d4305cec.json:
```diff
@@ -0,0 +1,26 @@
+        "name": "Multiple users identified in an RDP session",
+        "query": "process.Ext.authentication_id :* and process.Ext.session_info.client_address:* and host.ip:* and process.Ext.session_info.logon_type:\"RemoteInteractive\"  and event.category :\"process\"",
+        "threshold": {"field":["process.Ext.authentication_id"],"value":2,"cardinality":[{"field":"user.name","value":2}]},
+        "risk_score": 47,
```
Probably want to check the risk score/severity of the non-ML rules with the TRADE team if you haven't already.

sodhikirti07 replied:
Yes, I will cross check with someone on the team.

ajosh0504 reviewed (commented):
Interesting work!

Left some comments- some might be repetitive. Mainly (if you haven't already):
- Check the implications of including multiple detectors in a single job vs having separate jobs for each of those conditions.
- Try to focus on modeling by users/hosts (or close proxies to these entities) since those are the entities we're mainly concerned with.
- Use influencers as a place to provide as much context as possible about the anomaly.
- Double-check that the bucket spans and rule schedules are accurate. I see the transforms, jobs, and rules running on several different schedules so want to make sure you're not missing important activity due to misaligned schedules.

susan-shu-c commented on packages/lmd/docs/README.md:
```diff
@@ -1,6 +1,6 @@
 # Lateral Movement Detection Model
 
-The Lateral movement detection model package contains assets that detect lateral movement based on file transfer activity. This package requires a Platinum subscription. Please ensure that you have a Trial, Platinum, or Enterprise subscription before proceeding. This package is licensed under Elastic License v 2.0.
+The Lateral movement detection model package contains assets that detect lateral movement based on file transfer activity and Window's RDP events. This package requires a Platinum subscription. Please ensure that you have a Trial, Platinum, or Enterprise subscription before proceeding. This package is licensed under Elastic License v 2.0.
```
```suggestion
The Lateral movement detection model package contains assets that detect lateral movement based on file transfer activity and Window's RDP events. This package requires a Platinum subscription. Please ensure that you have a Trial, Platinum, or Enterprise subscription before proceeding. This package is licensed under [Elastic License 2.0](https://www.elastic.co/licensing/elastic-license).
```

susan-shu-c replied:
Add link and remove the `v` to make it aligned with the wording on our website

susan-shu-c commented on packages/lmd/docs/README.md:
```diff
@@ -9,6 +9,13 @@ To download the assets, click **Settings** > **Install Lateral Movement Detectio
 
 ### Add preconfigured anomaly detection jobs
 
+The anomaly detection jobs under this package rely on two indices. One has file transfer events (`logs-*`) and another index (`ml-rdp-lmd-1.0.2`) that collects RDP session information from a transform. Before enabling the jobs, create a data view with both index patterns.
```
```suggestion
The anomaly detection jobs under this package rely on two indices. One has file transfer events (`logs-*`), and the other index (`ml-rdp-lmd-1.0.2`) collects RDP session information from a transform. Before enabling the jobs, create a data view with both index patterns.
```

peteharverson commented on packages/lmd/kibana/ml_module/lmd-ml.json:
```diff
@@ -169,6 +152,245 @@
+                            ],
+                        "influencers": [
+                            "host.name",
+                            "user.name"
```
You probably want to include `destination.ip` as an influencer.

peteharverson commented on packages/lmd/docs/README.md:
```diff
@@ -1,6 +1,6 @@
 # Lateral Movement Detection Model
 
-The Lateral movement detection model package contains assets that detect lateral movement based on file transfer activity. This package requires a Platinum subscription. Please ensure that you have a Trial, Platinum, or Enterprise subscription before proceeding. This package is licensed under Elastic License v 2.0.
+The Lateral movement detection model package contains assets that detect lateral movement based on file transfer activity and Window's RDP events. This package requires a Platinum subscription. Please ensure that you have a Trial, Platinum, or Enterprise subscription before proceeding. This package is licensed under [Elastic License 2.0](https://www.elastic.co/licensing/elastic-license).
```
This should be without the apostrophe AFAIK - `Windows`

peteharverson commented on packages/lmd/manifest.yml:
```diff
@@ -1,16 +1,16 @@
 source:
   license: "Elastic-2.0"
-description: "ML package to detect lateral movement based on file transfer activity."
+description: "ML package to detect lateral movement based on file transfer activity and Window's RDP events."
```
As above, this should be without the apostrophe AFAIK - `Windows`

peteharverson commented on packages/lmd/docs/README.md:
```diff
@@ -21,24 +28,38 @@ This model uses both anomaly detection and security rules to detect lateral move
-| rare-file-path-remote-transfer | A machine learning job to detect unusual folders and directories on which a file is transferred (by a host) |
+| Job | Description                                                                                                              |
+|---|--------------------------------------------------------------------------------------------------------------------------|
+| high-count-remote-file-transfer | A machine learning job to detect unusually high file transfers to a remote host in the network                           | 
```
Nit - the second half of the list uses a closing `.`, while the first half doesn't.

peteharverson commented on packages/lmd/kibana/ml_module/lmd-ml.json:
```diff
@@ -169,6 +152,245 @@
+                        "security",
+                        "lateral_movement"
+                    ],
+                    "description": "A machine learning job to detect abnormally long RDP session duration.",
```
You drop `A machine learning job to...` for the anomaly detection job descriptions to make them more concise (which would look better in the UI), for example `Detects abnormally long RDP session duration.`

The format used in most of the other security modules includes the module name at the start - not sure if you want to follow that pattern e.g.

```
"Security: Packetbeat - Looks for unusual DNS activity that could indicate command-and-control or data exfiltration activity."
```

```
Security: Linux - Looks for commands related to system network connection discovery from an unusual user context. 
```

sodhikirti07 replied:
For now, I made the description more concise without including the module name.

peteharverson reviewed (commented):
A great update to this package! Just left a few comments mostly related to the ML jobs.

sodhikirti07 commented:
### Updates to the PR

Made below changes to resolve comments:
- Split the ml job `high-mean-var-rdp-session-duration` into `high-var-rdp-session-duration` and `high-mean-rdp-session-duration`. Updated datafeed for the two jobs
- Changed the `bucket_span` of ml jobs to `6h`. Added `source.ip` and `destination.ip` as influencers.
- Updated dashboard query filter to include new ml job
- Changed detection rule lookback time to 2*bucket_span of the ml job (now-12h)
- Deleted two rule based detection rules after consulting with TRADE again. Reason being such behavior is only seen for terminal servers and common Windows server won't allow more than 1 user per RDP session.

peteharverson reviewed (commented):
Latest changes for the ML jobs LGTM. Just left one query on the choice of influencers for the `unusual-time-weekday-rdp-session-start` job.

sodhikirti07 commented on packages/lmd/kibana/ml_module/lmd-ml.json:
```diff
@@ -169,6 +156,281 @@
+                        "influencers": [
+                            "host.name",
+                            "user.name",
+                            "destination.ip"
```
Will add both `source.ip` and `destination.ip` as I need information on both.

sodhikirti07 commented:
@ajosh0504 I added [instructions](https://github.com/elastic/integrations/blob/b9b4810e439053cbbd151f57ecf1ea98abe74764/packages/lmd/docs/README.md) to install and use `ProblemChild` with this package. Could you take a look and let me know if it's sufficient or require more detail?

ajosh0504 commented on packages/lmd/kibana/ml_module/lmd-ml.json:
```diff
@@ -109,7 +96,7 @@
-                    "description": "A machine learning job to detect rare file extensions shared with a remote host in the network.",
+                    "description": "Detects rare file extensions shared with a remote host in the network.",
                     "analysis_config": {
                         "bucket_span": "15m",
```
Isn't 15m too small?

sodhikirti07 replied:
For rare jobs it's recommended to keep the interval short. With a larger bucket_span the entities may seem to be less rare. 
https://www.elastic.co/guide/en/machine-learning/current/ml-rare-functions.html

ajosh0504 commented on packages/lmd/kibana/ml_module/lmd-ml.json:
```diff
@@ -169,6 +156,282 @@
                         "created_by": "ml-module-lmd"
```
Isn't 15m too small?

sodhikirti07 replied:
same as above

ajosh0504 commented on packages/lmd/kibana/ml_module/lmd-ml.json:
```diff
@@ -2,36 +2,23 @@
+            }},
+            {
+              "bool": {"must": [
+                {"exists": {"field": "session.start_time"}},
```
You could only look for one of these, for consistency with the datafeed queries.

ajosh0504 commented on packages/lmd/kibana/security_rule/462dfff5-6d66-4489-9d1d-f3b352b83665.json:
```diff
@@ -0,0 +1,26 @@
+        ],
+        "description": "An anomaly detection job has detected a high count of destination IPs establishing an RDP connection with a single source IP.\n",
+        "from": "now-12h",
+        "interval": "15m",
```
What does `interval` do?

sodhikirti07 replied:
`Interval` is the `Runs every` and `from` is the `Additional look-back time`
<img width="308" alt="image" src="https://github.com/elastic/integrations/assets/109447885/d0d73d66-4063-41d5-9fab-642e522db1b3">

sodhikirti07 commented on packages/lmd/docs/README.md:
```diff
@@ -9,36 +10,120 @@ To download the assets, click **Settings** > **Install Lateral Movement Detectio
+To detect malicious RDP processes started in a session, install the [Living off the Land Attack (LotL) Detection package](https://docs.elastic.co/integrations/problemchild). Follow the steps under the package [overview](https://docs.elastic.co/integrations/problemchild) to install related assets. Use the below filter query to examine model predictions on RDP events only. 
+
+Follow the below steps to customize anomaly detection jobs for RDP events:
+1. Edit the datafeed of anomaly detection jobs to include RDP process events only. Click on the **Actions** panel at the right-most corner of the job and then select the **Edit job** option.
```
Good point, will add the names here.

sodhikirti07 replied:
They will need to edit the datafeed of all jobs to only process RDP based events. I will specify it in the instructions.

ajosh0504 replied:
Ahhh go tit. Yeah, good to clarify that.

ajosh0504 commented on packages/lmd/docs/README.md:
```diff
@@ -9,36 +10,120 @@ To download the assets, click **Settings** > **Install Lateral Movement Detectio
 
+### Install ProblemChild package to detect malicious processes
+
+To detect malicious RDP processes started in a session, install the [Living off the Land Attack (LotL) Detection package](https://docs.elastic.co/integrations/problemchild). Follow the steps under the package [overview](https://docs.elastic.co/integrations/problemchild) to install related assets. Use the below filter query to examine model predictions on RDP events only. 
```
Do we want to link to the blog instead, since that has more detailed installation instructions.

sodhikirti07 replied:
I agree that the blog post has more detail, but the instructions need to be updated as we no longer use the detection rules repo to install the packages. It could be misleading for new users. The blog post is linked to the overview page, so it's still accessible. Does it make sense?

ajosh0504 reviewed (approved):
No further comments. LGTM! Nice work, @sodhikirti07! I learnt a lot from your PR. :)

stevewritescode commented on packages/lmd/docs/README.md:
```diff
@@ -9,36 +10,120 @@ To download the assets, click **Settings** > **Install Lateral Movement Detectio
 
 This model uses both anomaly detection and security rules to detect lateral movement in the network. In order to see all alerts detected by this model, you need to enable all the "Security Detection Rules" in the table, as described below. The first four rules are triggered when certain conditions for the anomaly detection jobs are satisfied. The last two rules are behavioral and independent of a ...
 
+### Install ProblemChild package to detect malicious processes
```
@sodhikirti07 if the user wants to use the LotL detection package to also detect non-lateral movement cases, can they still do that after making the modifications described in this section? Or is it an either/or decision between general LotL detection and specific RDP-based lateral movement detection using ProblemChild?

sodhikirti07 replied:
They can still see the ProblemChild predictions for all the process events including RDP. However, if they want to only see predictions for RDP, they can use the query to filter.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/53)
* Commit: fa428232cec103ee43114d2a9e4bafd6b671131a

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/53#0188b5db-6dc2-4b0a-ad9c-a44d911fc5d3)

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"running","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/53","number":53,"commit":"fa428232cec103ee43114d2a9e4bafd6b671131a"}],"number":53}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/60)
* Commit: 9d912260fb09c4bd5f7ea90e8f3d8a8dc31104ca

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/60#0188b624-badc-4aac-b0a9-2be86e5786ca)

### History
* :broken_heart: [Build #53](https://buildkite.com/elastic/integrations/builds/53) failed fa428232cec103ee43114d2a9e4bafd6b671131a

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/60","number":60,"commit":"9d912260fb09c4bd5f7ea90e8f3d8a8dc31104ca"},{"buildStatus":{"state":"running","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/53","number":53,"commit":"fa428232cec103ee43114d2a9e4bafd6b671131a"}],"number":60}
buildkite-pr-comment-->

peteharverson commented on packages/lmd/docs/README.md:
```diff
@@ -9,36 +10,120 @@ To download the assets, click **Settings** > **Install Lateral Movement Detectio
+4. Use the new data view (`logs-*, ml-rdp-lmd-1.0.2`) to create anomaly detection jobs for this package.
+
+
+In **Machine Learning > Anomaly Detection**, when you create a job, you should see an option to `Use preconfigured jobs` with a card for Lateral Movement Detection. When you select the card, you will see pre-configured anomaly detection jobs that you can enable depending on what makes the most sense for your environment.
```
One bit of feedback we've had from someone using the current version of this integration is that we could make it clearer that you only see the card inside the ML Anomaly Detection job creation wizard when you have docs in the selected data view which match the query defined in the `lmd-ml.json` file, which in this case is now:
```
          "bool": {"should": [
            {"term": {
              "event.category": "file"
            }},
            {
              "bool": {"filter": [
                {"exists": {"field": "session.start_time"}}
              ]}
            }
          ]}
```

Any thoughts on whether it is worth adding that sort of info?

peteharverson replied:
LGTM. Thanks for adding the note for the ML jobs query in the README.

peteharverson reviewed (approved):
I added an extra comment with some feedback from someone who's been using the current version of the integration, but otherwise the changes you've made for the ML jobs LGTM.

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/101)
* Commit: 4199cf03dab3c2acdf2dfb8d4bc3b88e5f8c97a2

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/101#0188ba1f-a845-4126-a9c5-0abd500cc080)

### History
* :broken_heart: [Build #60](https://buildkite.com/elastic/integrations/builds/60) failed 9d912260fb09c4bd5f7ea90e8f3d8a8dc31104ca
* :broken_heart: [Build #53](https://buildkite.com/elastic/integrations/builds/53) failed fa428232cec103ee43114d2a9e4bafd6b671131a

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/101","number":101,"commit":"4199cf03dab3c2acdf2dfb8d4bc3b88e5f8c97a2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/60","number":60,"commit":"9d912260fb09c4bd5f7ea90e8f3d8a8dc31104ca"},{"buildStatus":{"state":"running","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/53","number":53,"commit":"fa428232cec103ee43114d2a9e4bafd6b671131a"}],"number":101}
buildkite-pr-comment-->
